test/database_service/test_db_experiment_data.py

```diff
@@ -1201,3 +1200,90 @@ def test_hgp_setter(self):
         self.assertEqual("ibm-q-internal", exp_data.hub)
         self.assertEqual("deployed", exp_data.group)
         self.assertEqual("default", exp_data.project)
+
+    def test_add_delete_artifact(self):
+        """Tests adding an artifact and a list of artifacts. Tests deleting an artifact
+        by name, ID, and index and ID. Test the metadata is correctly tracking additions
+        and deletions."""
+        exp_data = ExperimentData()
+        self.assertEqual(exp_data.artifacts(), [])
+        new_artifact = ArtifactData(name="test", data="foo")
+        exp_data.add_artifacts(new_artifact)
+        self.assertEqual(exp_data.artifacts(0), new_artifact)
+        self.assertEqual(exp_data.artifacts("test"), new_artifact)
+
+        service = mock.create_autospec(IBMExperimentService, instance=True)
+        exp_data.service = service
+        exp_data.save()
+
+        self.assertEqual(exp_data.metadata["artifact_files"], {"test.zip"})
+
+        # delete by name
+        exp_data.delete_artifact("test")
+        self.assertEqual(exp_data.artifacts(), [])
+        self.assertEqual(exp_data._deleted_artifacts, {"test"})
```

Comment:
Do we want to test against protected method? This means we need to maintain the behavior of this method.

Reply:
Yeah it's not meant for external use, just like `exp_data._deleted_figures`, but I want to test that it has the correct behavior because there's no easy way to check that deletion worked correctly on the fake service (I think). We can update the test when the fake service is updated.